ember-server/src/test/scala/org/http4s/ember/server/ConnectionSuite.scala

```diff
@@ -64,7 +64,7 @@ class ConnectionSuite extends Http4sSuite {
       .withRequestHeaderReceiveTimeout(headerTimeout)
       .build
 
-  case class TestClient(client: Socket[IO]) {
+  sealed case class TestClient(client: Socket[IO]) {
```

Comment:
I think that case classes should be `final` rather than `sealed`.
`sealed` is needed for exhaustiveness checking on its subclasses, I guess.

Reply:
Uh oh, that's a test suite. So my comment is non-blocking at all.

Reply:
Also, final case classes that are members of another class omit a compiler warning that I don’t fully understand.